src/core/layertree/qgslayertreegroup.h

```diff
@@ -55,6 +55,21 @@ class CORE_EXPORT QgsLayerTreeGroup : public QgsLayerTreeNode
      */
     void setName( const QString &n ) override;
 
+    /**
+     * Returns the group's id.
+     */
+    QString id() const;
+
+    /**
+     * Sets the group's id.
+     */
+    void setId( const QString &id );
+
+    /**
+     * Generate the group's id, according to the groups's name (based on QgsMapLayer::generateId)
+     */
+    QString generateId( QString name ) const;
```

Comment:
Does it make any sense to have this public?

Reply:
Probably not, I moved the method to `protected`